src/AuthenticationService.php

```diff
@@ -492,8 +493,7 @@ public function isImpersonating(ServerRequestInterface $request): bool
      */
     protected function getImpersonationProvider(): ImpersonationInterface
     {
-        /** @var \Authentication\Authenticator\ImpersonationInterface $provider */
-        $provider = $this->getAuthenticationProvider();
+        $provider = $this->getAuthenticationProvider() ?? new stdClass();
```

Comment:
This is silly. Not need to create a `stdClass` instance if `null` is returned. Instead change the next line to `if ($provider === null)`  and the code after as necessary.

Reply:
I tried that but phpstan doesn't like having `get_class` getting passed a `null` 
Therefore I added a separate if with a separate exception.